tracer/tracer.go

```diff
@@ -266,10 +266,20 @@ func NewTracer(ctx context.Context, cfg *Config) (*Tracer, error) {
 		return nil, fmt.Errorf("failed to read kernel symbols: %v", err)
 	}
 
+	// Loading specifications about eBPF programs and maps from the embedded elf file
+	// does not load them into the kernel.
+	// A collection specification holds the information about eBPF programs and maps.
+	// References to eBPF maps in the eBPF programs are just placeholders that need to be
+	// replaced by the actual loaded maps later on with RewriteMaps before loading the
+	// programs into the kernel.
+	coll, err := support.LoadCollectionSpec(cfg.DebugTracer)
```

Comment:
Just curious, why did we lift this outside of `initializeMapsAndPrograms`, we don't currently use it anywhere else.